src/configMaker.js

```diff
@@ -12,15 +15,39 @@ const makeConfig = (vars) => {
     }).join(".");
 
     const configObject = {};
-    _.set(configObject, assignPath, vars[configKey]);
+    set(configObject, assignPath, vars[configKey]);
 
     configObjectsArray.push(configObject);
     return true;
   });
 
-  return _.merge({}, ...configObjectsArray);
+  return merge({}, ...configObjectsArray);
+};
+
+/* Build the full application configuration.
+ */
+const buildConfig = (name, defaults, vars, debug = false, testing = false) => {
+  const injector = getInjector();
+
+  // load environment variables
+  const environ = makeConfig(vars);
+
+  // generate metadata
+  const metadata = new Metadata(
+    name,
+    debug,
+    testing,
+  );
+
+  const config = merge(defaults, environ);
```

Comment:
This doesn't come through in the diff, but in the original code this was copied from, we were treating "metadata" like config and merging it into the config object. Ultimately I think I want to create a `graph` object from our injector and let the graph have both `metadata` and `config`. For now, just adding metadata as a separate key under config.